daemon/redis.c

```diff
@@ -1734,37 +1386,21 @@ static int json_link_maps(struct redis *r, struct call *c, struct redis_list *ma
 	return 0;
 }
 
-static int redis_link_maps(struct redis *r, struct call *c, struct redis_list *maps,
-		struct redis_list *sfds)
-{
-	unsigned int i;
-	struct endpoint_map *em;
-
-	for (i = 0; i < maps->len; i++) {
-		em = maps->ptrs[i];
-
-		if (redis_build_list_cb(&em->intf_sfds, r, "map_sfds", &c->callid, em->unique_id, sfds,
-					rbl_cb_intf_sfds, em))
-			return -1;
-	}
-	return 0;
-}
-
 static void json_restore_call(struct redis *r, struct callmaster *m, const str *callid, enum call_type type) {
 	redisReply* rr_jsonStr;
 	struct redis_hash call;
 	struct redis_list tags, sfds, streams, medias, maps;
 	struct call *c = NULL;
+	str s, id;
+
 	const char *err = 0;
 	int i;
-	str s;
 	JsonReader *root_reader =0;
 	JsonParser *parser =0;
 
-	rr_jsonStr = redis_get(r, REDIS_REPLY_STRING, "GET json-" PB, STR(callid));
+	rr_jsonStr = redis_get(r, REDIS_REPLY_STRING, "GET " PB, STR(callid));
 	if (!rr_jsonStr) {
-		rlog(LOG_ERR, "Could not retrieve json data from redis for callid: " STR_FORMAT,
-				STR_FMT(callid));
+		rlog(LOG_ERR, "Could not retrieve json data from redis for key: %s", callid->s);
```

Comment:
Please always use the `STR_FORMAT` and `STR_FMT()` macros when dealing with `str` objects. There's no inherent guarantee that the `->s` buffer is null terminated.